src/main/java/com/likelion/backendplus4/talkpick/batch/rss/service/RssService.java

```diff
@@ -0,0 +1,134 @@
+ */
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class RssService {
+
+    private final RssSourceConfig rssSourceConfig;
+    private final RssNewsRepository rssNewsRepository;
+    private final RssMappingFactory rssMappingFactory;
+
+    /**
+     * 모든 활성화된 RSS 뉴스 피드를 가져와 처리 (서비스 계층 메인기능)
+     *
+     * 1. 설정에서 활성화된 모든 RSS 소스 조회
+     * 2. 각 소스에서 RSS 피드 가져오기
+     * 3. 수집된 모든 뉴스 항목 병합
+     * 4. 데이터베이스에 저장 (중복 제외)
+     *
+     * @return 새로 저장된 뉴스 항목 수
+     * @throws Exception RSS 처리 중 발생할 수 있는 예외
+     * @since 2024-05-10 최초 작성
+     * @author 양병학
+     * @modify 2024-05-10 양병학
+     */
+    @Transactional
+    public int fetchAndProcessAllFeeds() {
+        List<RssNews> allNewsItems = new ArrayList<>();
+
+        rssSourceConfig.getSources().stream()
+                .filter(RssSourceConfig.RssSource::isEnabled)
+                .forEach(source -> {
+                    try {
+                        List<RssNews> items = fetchFeedFromSource(source);
+                        allNewsItems.addAll(items);
+                    } catch (Exception e) {
+                        log.error("Error: 뉴스피드 fetch중 {}: {}", source.getName(), e.getMessage(), e);
+                    }
+                });
+
+        return saveItems(allNewsItems);
+    }
+
+    /**
+     * 특정 소스에서 RSS 피드를 가져옴
+     *
+     * 1. 소스 URL(application-dev)에서 뉴스 피드 로드 (dev파일은 디스코드 스레드에 올리겠습니다)
+     * 2. 소스 유형에 맞는 Mapper(server/mapper) 가져오기
+     * 3. 각 항목을 RssNews 엔티티로 변환
+     *
+     * @param source RSS 피드 소스 설정
+     * @return 변환된 RssNews 엔티티 목록
+     * @throws Exception RSS 피드 로드 및 처리 중 발생할 수 있는 예외
+     * @since 2024-05-10 최초 작성
+     * @author 양병학
+     */
+    private List<RssNews> fetchFeedFromSource(RssSourceConfig.RssSource source) throws Exception {
+        URL feedUrl = new URL(source.getUrl());
+        SyndFeedInput input = new SyndFeedInput();
+        SyndFeed feed = input.build(new XmlReader(feedUrl));
+
+        List<RssNews> newsItems = new ArrayList<>();
+        RssMapper mapper = rssMappingFactory.getMapper(source.getType());
+
+        for (SyndEntry entry : feed.getEntries()) {
+            RssNews newsItem = mapper.mapToRssNews(entry);
+            newsItems.add(newsItem);
+        }
+
+        log.info("페치완료 {} 갯수 뉴스피드 {} 에서", newsItems.size(), source.getName());
+        return newsItems;
+    }
+
+    /**
+     * 수집된 뉴스 항목을 DB에 저장합니다.(추후 Batch로 다시구현하면 없어질 메서드)
+     * 중복된 항목은 건너뛰고 새로운 항목만 저장. (INSERT IGNORE로 대체시 Native Query 필요, 수정사항 처리 메서드 아직 없음)
+     *
+     * 1. 기존 존재하는 링크 필터링
+     * 2. 새 항목만 데이터베이스에 저장
+     * 3. 중복 저장 시도 시 예외 처리
+     *
+     * @param newsItems 저장할 뉴스 항목 목록
+     * @return 새로 저장된 항목 수
+     * @since 2024-05-10 최초 작성
+     * @author 양병학
+     * @modify 2024-05-10 양병학
+     */
+    @Transactional
+    public int saveItems(List<RssNews> newsItems) {
+        int savedCount = 0;
+
+        List<RssNews> newItems = newsItems.stream()
+                .filter(item -> !rssNewsRepository.existsByLink(item.getLink()))
```

Comment:
코드 가독성 향상을 위해 이렇게 수정해보는건 어떨까요?

```suggestion
                .filter(not(item -> rssNewsRepository.existsByLink(item.getLink())))
```

아니면, 나중에 JPA 어댑터에서 notExist 메소드를 만들어주면 좋겠습니다..

Reply:
가독성으로 보면 첫번째 제안이 좋은거 같아서 반영하겠습니다
두번째 제안이신 notExist도 좋지만 헥사고날로 리팩토링하면서 아웃바운드에 그떄 추가하는게 편하지않을까 해서
일반 첫번째 제안을 반영하겠습니다.